### PR TITLE
Move from garyburd Redigo to supported redigo

### PR DESCRIPTION
**Pull request by verajohne**
This PR moves from Garyburd Redigo to go module redigo. It also explicitly sets InsecureSkipVerify on the tls config as this was only set if tlsConfig was not set. See [redigo code](https://github.com/gomodule/redigo/blob/72af8129e040d6f962772a8c582e5e9f22085788/redis/conn.go#L246)

### CODE REVIEW

File: internal/peer/redis.go [hunk 1]
```diff
@@ -179,11 +179,11 @@ func buildOptions(c config.Config) []redis.DialOption {
 	useTLS, _ := c.GetUseTLS()
 	if useTLS {
 		tlsConfig := &tls.Config{
-			MinVersion: tls.VersionTLS12,
+			MinVersion:         tls.VersionTLS12,
+			InsecureSkipVerify: true,
```
Comment (vreynolds) [inline internal/peer/redis.go:183]:
```suggestion
```

Thanks for catching that skip verify was a noop! We came to the conclusion that it ought not to have been set in the first place, so let's just remove the noop line, and allow it to keep verifying as it is doing today.

**Review by vreynolds — CHANGES_REQUESTED** [review]
Thank you! See comment about TLS skip verify, otherwise LGTM

File: internal/peer/redis.go (hunk 1 above)
Comment (vreynolds) [reply internal/peer/redis.go:183]: Ah, understood. We thought it was just something ya'll noticed while bumping the dependency. If you'd like, we can just merge the dependency change in this PR, and open a new issue for the skipVerify? On the issue, can you provide some context for your use case?

Comment (verajohne) [reply internal/peer/redis.go]: @vreynolds Removed the code for InsecureSkipVerify from this PR. And will open an issue

**Review by vreynolds — APPROVED** [review]

**Pull request merged**